File: backend/src/main/java/com/settleup/domain/settlement/Settlement.java
package com.settleup.domain.settlement;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

/**
 * Settlement Entity
 * 여행 정산 또는 게임 정산 세션
 */
@Entity
@Table(name = "settlements")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Settlement {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @NotBlank(message = "제목은 필수입니다")
    @Size(min = 1, max = 100, message = "제목은 1-100자 사이여야 합니다")
    @Column(nullable = false, length = 100)
    private String title;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private SettlementType type;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private SettlementStatus status = SettlementStatus.ACTIVE;

    @Column(name = "creator_id", nullable = false)
    private UUID creatorId;

    @Size(max = 500, message = "설명은 최대 500자입니다")
    @Column(length = 500)
    private String description;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(nullable = false, length = 3)
    @Builder.Default
    private String currency = "KRW";

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @Version
    @Builder.Default
    private Integer version = 0;

    @Column(name = "sync_status", nullable = false, length = 20)
    @Builder.Default
    private String syncStatus = "SYNCED";

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (status == null) {
            status = SettlementStatus.ACTIVE;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    /**
     * 종료일이 시작일 이후인지 검증
     */
    public void validateDates() {
        if (endDate != null && startDate != null && endDate.isBefore(startDate)) {
            throw new IllegalArgumentException("종료일은 시작일 이후여야 합니다");
        }
    }
}

File: mobile/src/models/Settlement.ts
/**
 * Settlement 모델
 * 정산 데이터 구조 정의
 */

/**
 * 정산 타입 (여행 또는 게임)
 */
export enum SettlementType {
  TRAVEL = 'TRAVEL',
  GAME = 'GAME',
}

/**
 * 정산 상태
 */
export enum SettlementStatus {
  ACTIVE = 'ACTIVE',         // 진행 중
  COMPLETED = 'COMPLETED',   // 완료됨
  ARCHIVED = 'ARCHIVED',     // 보관됨
}

/**
 * 정산 인터페이스
 */
export interface Settlement {
  /** 정산 ID (UUID) */
  id: string;

  /** 정산 제목 */
  title: string;

  /** 정산 타입 */
  type: SettlementType;

  /** 정산 상태 */
  status: SettlementStatus;

  /** 생성자 ID (사용자 ID) */
  creatorId: string;

  /** 정산 설명 (선택적) */
  description?: string;

  /** 시작 날짜 (ISO 8601: YYYY-MM-DD) */
  startDate?: string;

  /** 종료 날짜 (ISO 8601: YYYY-MM-DD) */
  endDate?: string;

  /** 통화 코드 (기본값: KRW) */
  currency: string;

  /** 생성 일시 (ISO 8601: YYYY-MM-DDTHH:mm:ss) */
  createdAt: string;

  /** 수정 일시 (ISO 8601: YYYY-MM-DDTHH:mm:ss) */
  updatedAt: string;

  /** 총 지출액 (요약) */
  totalExpense?: number;

  /** 활성 참가자 수 (요약) */
  participantCount?: number;

  /** 게임 라운드 수 (요약) */
  roundCount?: number;
}

/**
 * 정산 생성 요청 데이터
 */
export interface CreateSettlementRequest {
  title: string;
  type: SettlementType;
  description?: string;
  startDate?: string;
  endDate?: string;
  currency?: string;
}

/**
 * 정산 업데이트 요청 데이터
 */
export interface UpdateSettlementRequest {
  title?: string;
  description?: string;
  startDate?: string;
  endDate?: string;
  status?: SettlementStatus;
  currency?: string;
}
